service/host_mock.go

```diff
@@ -41,7 +41,7 @@ func NullHost() Host {
 			log: ulog.NoopLogger,
 		},
 		metricsCore: metricsCore{
-			scope: tally.NoopScope,
+			metrics: tally.NoopScope,
```

Comment:
add a null stats reporter here as well, please otherwise some tests might crash due to runtime metrics trying to use it